### PR TITLE
STCC-35: Extension to show variables

### PR DESCRIPTION
**Pull request by derfloh205**

### CODE REVIEW

File: src/scratchtocatrobat/converter/converter.py [hunk 1]
```diff
@@ -871,13 +871,28 @@ def _catrobat_sound_from(scratch_sound):
...
+    @staticmethod
+    def _create_show_text_brick_and_update_positions(var_object, context):
+        show_variable_brick = catbricks.ShowTextBrick(context.visible_var_X, context.visible_var_Y)
+        show_variable_brick.setUserVariableName(var_object.getName())
+        show_variable_brick.setUserVariable(var_object)
+        context.visible_var_Y -= VISIBLE_VAR_POSITION_STEP_Y
+        if context.visible_var_Y <= VISIBLE_VAR_POSITION_THRESHOLD_Y:
+            context.visible_var_Y = VISIBLE_VAR_Y_INIT
+            context.visible_var_X += VISIBLE_VAR_POSITION_STEP_X
+        if context.visible_var_X >= VISIBLE_VAR_POSITION_THRESHOLD_X:
```
Comment (cfkh) [inline src/scratchtocatrobat/converter/converter.py]: Shouldn't also context.visible_var_Y be compared to a threshold?

File: src/scratchtocatrobat/converter/converter.py [hunk 2]
```diff
@@ -871,13 +871,28 @@ def _catrobat_sound_from(scratch_sound):
...
+                log.info("Too many visible variables")
```
Comment (cfkh) [inline src/scratchtocatrobat/converter/converter.py]: The log can't be seen by users on their mobile devices. You could possibly additionally add a Note Brick with an appropriate warning to the script.

File: src/scratchtocatrobat/converter/converter.py [hunk 3]
```diff
@@ -871,13 +871,30 @@ def _catrobat_sound_from(scratch_sound):
         sound_resource_name = scratch_sound[scratchkeys.SOUND_NAME]
         soundinfo.setSoundFileName(mediaconverter.catrobat_resource_file_name_for(sound_md5_filename, sound_resource_name))
         return soundinfo
```
Comment (rsamer) [inline src/scratchtocatrobat/converter/converter.py]: Missing new line

File: src/scratchtocatrobat/converter/converter.py [hunk 4]
```diff
@@ -974,28 +987,96 @@ def _add_default_behaviour_to(sprite, sprite_context, catrobat_scene, catrobat_p
...
+            local_sprite_commands = scratch_project._sprite_to_command_var_dict[sprite_name]
+        
+        commands_without_loop = ["answer", "timer"]
+        def isLoopNeeded():
+            for (command, _) in local_sprite_commands:
+                if command not in commands_without_loop:
+                    return True
+            return False
+        
+        if local_sprite_commands is not None and len(local_sprite_commands) >= 1 and isLoopNeeded():
```
Comment (rsamer) [inline src/scratchtocatrobat/converter/converter.py]: Naming convention: is_loop_needed

File: src/scratchtocatrobat/converter/converter.py [hunk 5]
```diff
@@ -974,28 +987,96 @@ def _add_default_behaviour_to(sprite, sprite_context, catrobat_scene, catrobat_p
...
+            loop_start = catbricks.ForeverBrick()
+            loop_end = catbricks.LoopEndBrick(loop_start)
+            loop_bricks = [loop_start, loop_end]
+            exclusive_start_script.getBrickList().addAll(0, loop_bricks)
+            wait_pos = len(exclusive_start_script.getBrickList()) - 1
+            wait_brick = catbricks.WaitBrick(100)
+            exclusive_start_script.getBrickList().addAll(wait_pos, [wait_brick])
```
Comment (rsamer) [inline src/scratchtocatrobat/converter/converter.py]: new line is missing

File: src/scratchtocatrobat/converter/converter.py [hunk 6]
```diff
@@ -974,28 +987,96 @@ def _add_default_behaviour_to(sprite, sprite_context, catrobat_scene, catrobat_p
...
+            match_count = 2
+            taken = True
+            test_name = name_to_check
+            while taken:
+                taken = _ScratchObjectConverter._variable_exists(test_name, catrobat_project) 
+                if taken:
+                    test_name = name_to_check + str(match_count)
+                    match_count += 1
+            return test_name
+        add_note_brick = False
```
Comment (rsamer) [inline src/scratchtocatrobat/converter/converter.py]: new line

File: src/scratchtocatrobat/converter/converter.py [hunk 7]
```diff
@@ -871,13 +871,30 @@ def _catrobat_sound_from(scratch_sound):
         sound_resource_name = scratch_sound[scratchkeys.SOUND_NAME]
         soundinfo.setSoundFileName(mediaconverter.catrobat_resource_file_name_for(sound_md5_filename, sound_resource_name))
         return soundinfo
+    @staticmethod
+    def _variable_exists(var, catrobat_project):
```
Comment (rsamer) [inline src/scratchtocatrobat/converter/converter.py]: Please move this *helper* function to the catrobat.py module.

**Comment by rsamer** [comment]
Unfortunately, all features of this PR are implemented in a wrong way.
As discussed in a previous meeting the goal of this task was to inject Scratch code into the parsed JSON code and to ask the frontend to rebuilt the object hierarchy (AST tree) (see timer workaround: `_add_timer_script_to_stage_object` method in `scratch.py` module).
This solution is more elegant, concise as well as requires less code for the same end result.
Thus, the bottom line is that I cannot merge this PR.

**Pull request closed without merging**